src/rust/e2e/src/utils.rs

```diff
@@ -0,0 +1,145 @@
+use std::{str::FromStr, sync::atomic::AtomicU64, time::Duration};
+
+use anyhow::Result;
+use aptos_sdk::{
+    bcs,
+    move_types::{
+        ident_str,
+        language_storage::{ModuleId, StructTag, TypeTag},
+    },
+    rest_client::{aptos_api_types::MoveModuleId, FaucetClient},
+    types::{account_address::AccountAddress, transaction::EntryFunction, LocalAccount},
+};
+use clap::Parser;
+use econia_sdk::EconiaClient;
+use sqlx::PgPool;
+
+pub const TIMEOUT: Duration = Duration::from_secs(1);
+
+#[derive(Parser, Debug)]
+pub struct Args {
+    /// The URL of the Aptos node
+    pub node_url: String,
+
+    /// The URL of the faucet
+    pub faucet_url: String,
+
+    /// The address of the Econia contract
+    pub econia_address: String,
+
+    /// The address of the Aptos faucet
+    pub faucet_address: String,
+
+    /// The database URL
+    pub db_url: String,
+}
+
+pub struct State {
+    pub e_apt: TypeTag,
+    pub e_usdc: TypeTag,
+    pub faucet_address: AccountAddress,
+    pub faucet_client: FaucetClient,
+    pub econia_address: AccountAddress,
+    pub econia_client: EconiaClient,
+    pub db_pool: PgPool,
+    pub node_url: String,
+    pub market_size: AtomicU64,
+}
+
+/// Creates the initial variables needed
+pub async fn init(args: &Args) -> State {
+    // Create eAPT and eUSDC `TypeTag`s
+    let faucet_address = if args.faucet_address.starts_with("0x") {
+        args.faucet_address.clone()
+    } else {
+        format!("0x{}", args.faucet_address)
+    };
+    let faucet_address =
+        AccountAddress::from_hex_literal(&faucet_address).expect("Could not parse faucet address.");
+    let e_apt = TypeTag::Struct(Box::new(
+        StructTag::from_str(&format!("0x{faucet_address}::example_apt::ExampleAPT")).unwrap(),
+    ));
+    let e_usdc = TypeTag::Struct(Box::new(
+        StructTag::from_str(&format!("0x{faucet_address}::example_usdc::ExampleUSDC")).unwrap(),
+    ));
+
+    // Create a `FaucetClient`
+    let faucet_client = FaucetClient::new(
+        reqwest::Url::parse(&args.faucet_url).unwrap(),
+        reqwest::Url::parse(&args.node_url).unwrap(),
+    );
+
+    // Transform the Econia address from `String` to `AccountAddress`
+    let econia_address = if args.econia_address.starts_with("0x") {
+        args.econia_address.clone()
+    } else {
+        format!("0x{}", args.econia_address)
+    };
+    let econia_address =
+        AccountAddress::from_hex_literal(&econia_address).expect("Could not parse Econia address.");
+
+    let (_, econia_client) = account(&faucet_client, &args.node_url, econia_address.clone()).await;
+
+    let db_pool = PgPool::connect(&args.db_url)
+        .await
+        .expect("Could not connect to the database.");
+
+    State {
+        e_apt,
+        e_usdc,
+        faucet_address,
+        faucet_client,
+        econia_address,
+        econia_client,
+        db_pool,
+        node_url: args.node_url.clone(),
+        market_size: AtomicU64::new(1),
```

Comment:
Hmmm I think you'll need to check the number of registered markets and start there (plus one). This makes it so that you can only run the tests once for a given chain instance, which I imagine we'll be running them multiple times as we change things.

Reply:
I added logic to query the min size on start.